dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerClient.java

```diff
@@ -199,7 +132,19 @@ private <T> T doHttpGet(String path, Map<String, String> params, Class<T> bodyCl
       }
       HttpEntity entity = httpResponse.getEntity();
       try {
-        return MAPPER.readValue(EntityUtils.toString(entity), bodyClass);
+        Map<String, Object> payload = MAPPER.readValue(EntityUtils.toString(entity), Map.class);
+        if (!payload.containsKey(payloadProperty)) {
+          throw new RangerException("Missing key " + payloadProperty + " in Ranger response");
+        }
+        Object items = payload.get(payloadProperty);
+        if (!(items instanceof List)) {
+          throw new RangerException(
+              "Expected an array value for key "
+                  + payloadProperty
+                  + " in Ranger response, got "
+                  + items.getClass());
+        }
+        return ImmutableList.copyOf((List<Object>) items);
```

Comment:
Should this be a `List<?>`, not a `List<Object>`, since we don't actually know what type is in it, but that type does exist, and is fixed?

Reply:
Done.